crates/polars-python/src/lazyframe/visitor/nodes.rs

```diff
@@ -470,26 +485,51 @@ pub(crate) fn into_py(py: Python<'_>, plan: &IR) -> PyResult<PyObject> {
-            options: (
-                match options.args.how {
-                    JoinType::Left => "left",
-                    JoinType::Right => "right",
-                    JoinType::Inner => "inner",
-                    JoinType::Full => "full",
-                    #[cfg(feature = "asof_join")]
-                    JoinType::AsOf(_) => return Err(PyNotImplementedError::new_err("asof join")),
-                    JoinType::Cross => "cross",
-                    JoinType::Semi => "leftsemi",
-                    JoinType::Anti => "leftanti",
+            options: {
+                let how = &options.args.how;
+                if how.is_ie() {
                     #[cfg(feature = "iejoin")]
-                    JoinType::IEJoin(_) => return Err(PyNotImplementedError::new_err("IEJoin")),
-                },
-                options.args.join_nulls,
-                options.args.slice,
-                options.args.suffix.as_deref(),
-                options.args.coalesce.coalesce(&options.args.how),
-            )
-                .to_object(py),
+                    if let JoinType::IEJoin(ie_options) = how {
+                        (
+                            "inequality",
+                            options.args.join_nulls,
+                            options.args.slice,
+                            options.args.suffix.as_deref(),
+                            options.args.coalesce.coalesce(&options.args.how),
+                            Wrap(ie_options.operator1).into_py(py),
+                            ie_options
+                                .operator2
+                                .as_ref()
+                                .map_or_else(|| py.None(), |op| Wrap(*op).into_py(py)),
+                        )
+                            .to_object(py)
+                    } else {
+                        unreachable!()
+                    }
+                } else {
+                    (
+                        match how {
+                            JoinType::Left => "left",
+                            JoinType::Right => "right",
+                            JoinType::Inner => "inner",
+                            JoinType::Full => "full",
+                            #[cfg(feature = "asof_join")]
+                            JoinType::AsOf(_) => {
+                                return Err(PyNotImplementedError::new_err("asof join"))
+                            },
+                            JoinType::Cross => "cross",
+                            JoinType::Semi => "leftsemi",
+                            JoinType::Anti => "leftanti",
+                            _ => unreachable!(),
+                        },
+                        options.args.join_nulls,
+                        options.args.slice,
+                        options.args.suffix.as_deref(),
+                        options.args.coalesce.coalesce(how),
+                    )
+                        .to_object(py)
```

Comment:
Not very happy with this absolute horror-show, suggestions gratefully received.

Reply:
Instead of returning `&str` and converting them implictly to object late, we can convert directly to objects and allow more complicated nesting, similar to the nesting of structs we have on the rust side.

Will adapt the PR.